packages/core/src/mutants/mutant-test-planner.ts

```diff
@@ -0,0 +1,211 @@
+import { Sandbox } from '../sandbox';
+
+import { MutantTestPlan } from './mutant-test-plan';
+
+import { PlanKind } from '.';
+
+/**
+ * The factor by which hit count from dry run is multiplied to calculate the hit limit for a mutant.
+ * This is intentionally a high value to prevent false positives.
+ *
+ * For example, a property testing library might execute a failing scenario multiple times to determine the smallest possible counterexample.
+ * @see https://jsverify.github.io/#minimal-counterexample
+ */
+const HIT_LIMIT_FACTOR = 100;
+
+/**
+ * Responsible for determining the tests to execute for each mutant, as well as other run option specific details
+ *
+ */
+export class MutantTestPlanner {
+  public static readonly inject = tokens(
+    coreTokens.dryRunResult,
+    coreTokens.reporter,
+    coreTokens.sandbox,
+    coreTokens.timeOverheadMS,
+    commonTokens.options,
+    commonTokens.logger
+  );
+  private readonly testsByMutantId: Map<string, Set<TestResult>>;
+  private readonly hitsByMutantId: Map<string, number>;
+  private readonly timeSpentAllTests: number;
+
+  constructor(
+    private readonly dryRunResult: CompleteDryRunResult,
+    private readonly reporter: StrictReporter,
+    private readonly sandbox: I<Sandbox>,
+    private readonly timeOverheadMS: number,
+    private readonly options: StrykerOptions,
+    logger: Logger
+  ) {
+    this.testsByMutantId = findTestsByMutant(this.dryRunResult.mutantCoverage?.perTest, dryRunResult.tests, logger);
+    this.hitsByMutantId = findHitsByMutantId(this.dryRunResult.mutantCoverage);
+    this.timeSpentAllTests = calculateTotalTime(this.dryRunResult.tests);
+  }
+
+  public makePlan(mutants: readonly Mutant[]): readonly MutantTestPlan[] {
+    const mutantTestPlans = mutants.map((mutant) => this.planMutant(mutant));
+    this.reporter.onAllMutantsMatchedWithTests(mutantTestPlans.map(({ mutant }) => mutant));
+    return mutantTestPlans;
+  }
+
+  private planMutant(mutant: Mutant): MutantTestPlan {
+    // If mutant was already ignored, pass that along
+    const isStatic = this.dryRunResult.mutantCoverage && this.dryRunResult.mutantCoverage.static[mutant.id] > 0;
+    if (mutant.status) {
+      // If this mutant was already ignored, early result
+      return {
+        plan: PlanKind.EarlyResult,
+        mutant: {
+          ...mutant,
+          status: mutant.status,
+          static: isStatic,
+        },
+      };
+    } else if (this.dryRunResult.mutantCoverage) {
+      // If there was coverage information (coverageAnalysis not "off")
+      const tests = this.testsByMutantId.get(mutant.id) ?? [];
+      const coveredBy = toTestIds(tests);
+      if (!isStatic || (this.options.ignoreStatic && coveredBy.length)) {
+        // If not static, or it was "hybrid" (both static and perTest coverage) and ignoreStatic is on.
+        // Only run covered tests
+        const estimatedNetTime = calculateTotalTime(tests);
+        return {
```

Comment:
It might be nice to have a factory/create method for these `MutantTestPlan` objects because current function is quite long

Reply:
Sure thing

Reply:
I've created factory methods for early result as well as for mutant plan